core/src/main/java/com/linecorp/armeria/server/logging/AccessLogFormats.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.logging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.linecorp.armeria.server.logging.AccessLogComponent.ofPredefinedCommon;
+import static com.linecorp.armeria.server.logging.AccessLogComponent.ofQuotedRequestHeader;
+import static com.linecorp.armeria.server.logging.AccessLogComponent.ofText;
+import static com.linecorp.armeria.server.logging.AccessLogType.AUTHENTICATED_USER;
+import static com.linecorp.armeria.server.logging.AccessLogType.REMOTE_HOST;
+import static com.linecorp.armeria.server.logging.AccessLogType.REQUEST_LINE;
+import static com.linecorp.armeria.server.logging.AccessLogType.REQUEST_TIMESTAMP;
+import static com.linecorp.armeria.server.logging.AccessLogType.RESPONSE_LENGTH;
+import static com.linecorp.armeria.server.logging.AccessLogType.RESPONSE_STATUS_CODE;
+import static com.linecorp.armeria.server.logging.AccessLogType.RFC931;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import com.linecorp.armeria.common.HttpHeaderNames;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.server.logging.AccessLogComponent.AttributeComponent;
+import com.linecorp.armeria.server.logging.AccessLogComponent.CommonComponent;
+import com.linecorp.armeria.server.logging.AccessLogComponent.RequestHeaderComponent;
+import com.linecorp.armeria.server.logging.AccessLogComponent.TextComponent;
+
+import io.netty.util.AsciiString;
+
+/**
+ * Pre-defined access log formats and the utility methods for {@link AccessLogComponent}.
+ */
+public final class AccessLogFormats {
```

Comment:
- This class could be package-local?
- How about creating a new public class like this:

  ```java
  public final class AccessLogWriters {
      public static Consumer<RequestLog> common() { ... }
      public static Consumer<RequestLog> combined() { ... }
      public static Consumer<RequestLog> disabled() { ... }
      private AccessLogWriters() {}
  }
  ```

  .. so that a user can just: `serverBuilder.accessLogWriter(AccessLogWriters.common())`?